event_model/tests/test_em.py

```diff
@@ -1196,3 +1196,47 @@ def test_pickle_filler():
     serialized = pickle.dumps(filler)
     deserialized = pickle.loads(serialized)
     assert filler == deserialized
+
+def test_projections():
+    event_model.schema_validators[event_model.DocumentNames.start].validate(
+    {
+        "uid": "abc",
+        "time": 0,
+        "projections": [
+            {
+                "name": "test",
+                "version": "42.0.0",
+                "configuration": {},
+                "projection": {
+                    'Entry/instrument/detector/data': {
```

Comment:
`/entry/instrument/detector/data`

Reply:
@prjemian ....does this reflect a consensus about how serialize pointers into fields in a NeXuS document? Are there other documented rules?

Reply:
Not sure I know how to answer that.  So, my answer has become verbose.

First off, the key of the projection dictionary is, I assume, the HDF5 address of the specific item in the **NeXus** file.  The NeXus [naming convention](https://manual.nexusformat.org/datarules.html#naming-conventions) is to use lower case.  I prepended a `/` since this should be an absolute address in the file.  Following NeXus convention for picking default names of groups [1], the path to a dataset `/entry/instrument/detector/data` names the `data` dataset of the NXdetector group (child of the NXinstrument group which is child of the NXentry group which is child of the file root).

[1] [default group names in NeXus](https://manual.nexusformat.org/rules.html#rules-for-structuring-information-in-nexus-files): For a hypothetical base class of `NXzzzz`, the default group would be named `zzzz` or `zzzz_1`, `zzzz_2` if there are several such.  In some NeXus classes, a specific name is required.  Mostly in the base classes, the name is more flexible.  Use the default naming unless you have a good reason to choose otherwise.

An [example is shown](https://apstools.readthedocs.io/en/latest/source/_filewriters.html#full-structure) in the documentation for a [NeXus filewriter callback](https://apstools.readthedocs.io/en/latest/source/_filewriters.html#hdf5-nexus-file-writers) I've written for the APS.  The example shows the raw data stored under `/entry/instrument/bluesky`. NeXus-style links position this data at various places in the NeXus structure satisfying the expectations of the NeXus base class structure.

Reply:
So our projection keys would point into the conventional NeXus base class structures.  A good test would be:

1. ingest from a NeXus file into databroker with declared projections
1. `nxwriter = apstools.filewriters.NXWriter()`
1. `for key, doc in db[-1]: nxwriter.receiver(key, doc)`
1. compare the new NeXus file with the original